source/module_lr/potentials/pot_hxc_lrtd.cpp

```diff
@@ -6,28 +6,69 @@
 #include "module_hamilt_general/module_xc/xc_functional.h"
 #include <set>
 #include "module_lr/utils/lr_util.h"
-
+#include "module_io/cube_io.h"
+#include "module_hamilt_pw/hamilt_pwdft/global.h"    // tmp, for pgrid
 #define FXC_PARA_TYPE const double* const rho, ModuleBase::matrix& v_eff, const std::vector<int>& ispin_op = { 0,0 }
 namespace LR
 {
     // constructor for exchange-correlation kernel
     PotHxcLR::PotHxcLR(const std::string& xc_kernel_in, const ModulePW::PW_Basis* rho_basis_in, const UnitCell* ucell_in,
-        const Charge* chg_gs/*ground state*/, const SpinType& st_in)
-        :xc_kernel(xc_kernel_in), tpiba_(ucell_in->tpiba), spin_type_(st_in)
+        const Charge* chg_gs/*ground state*/, const Parallel_Grid& pgrid,
+        const SpinType& st_in, const std::vector<std::string>& init_fxc)
+        :xc_kernel(xc_kernel_in), tpiba_(ucell_in->tpiba), spin_type_(st_in),
+        xc_kernel_components_(*rho_basis_in)
     {
         this->rho_basis_ = rho_basis_in;
         this->nrxx = chg_gs->nrxx;
         this->nspin = (PARAM.inp.nspin == 1 || (PARAM.inp.nspin == 4 && !PARAM.globalv.domag && !PARAM.globalv.domag_z)) ? 1 : 2;
 
         this->pot_hartree = LR_Util::make_unique<elecstate::PotHartree>(this->rho_basis_);
+
         std::set<std::string> local_xc = { "lda", "pbe", "hse" };
         if (local_xc.find(this->xc_kernel) != local_xc.end())
         {
             XC_Functional::set_xc_type(this->xc_kernel);    // for hse, (1-alpha) and omega are set here
```

Comment:
It will push_back something into `func_id` to a static variable here:
https://github.com/deepmodeling/abacus-develop/blob/edfa4c120b4edfb012d23fd1e16ef92cb868aaac/source/module_hamilt_general/module_xc/xc_functional.h#L76

Reply:
does it mean this will change the func_id (nearly a runtime environment variable in ABACUS)? is it safe?

Reply:
Yes it will change `func_id`.
 I think it's safe from the perspective of lr-tddft method because there's no other usage of func_id after the calculation of xc kernel.